### PR TITLE
fix(debian): allow version to be user-provided, don't force v13

### PR DESCRIPTION
**Pull request by Yoda-BZH**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

Yes. nodejs default version is set to 14, the LTS version.

### Describe the changes you're proposing

Debian repository is forced on version 13.

### Pillar / config required to test the proposed changes

https://github.com/saltstack-formulas/node-formula/blob/d8a8264ccaea147b65049b2cc9bd8473d1a74ebc/node/osfamilymap.yaml#L31

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by Yoda-BZH** [comment]
no news ?

**Comment by myii** [comment]
@Yoda-BZH Let's see if @noelmcloughlin can review this PR.

A few _related_ issues I've noticed:

1. There are no entries for `node:pkg:version` in `pillar.example` or even any of the test pillars -- I believe this should be added to at least the `pillar.example` and the `default.sls` test pillar.
1. If this is to be considered a breaking change, then the commit message should reflect that as [outlined here](https://github.com/saltstack-formulas/.github/blob/master/CONTRIBUTING.rst#use-breaking-change-to-trigger-a-major-version-change).

File: node/osfamilymap.yaml
Comment (noelmcloughlin) [inline node/osfamilymap.yaml:31]: I agree with this PR.  Could you add `node:pkg:version: '14'` to `defaults.yaml` so formula works without pillars?

**Review by noelmcloughlin — CHANGES_REQUESTED** [review]

File: node/osfamilymap.yaml
Comment (noelmcloughlin) [reply node/osfamilymap.yaml:31]: It should pick version from default.yaml unless (optional) pillars say different version.

Comment (Yoda-BZH) [reply node/osfamilymap.yaml:31]: changed, thank you

**Comment by noelmcloughlin** [comment]
LGTM but [Semantic-Release](https://github.com/semantic-release/semantic-release) is complaining about the commit message.
```
⧗   input: Set node:pkg:version to 14 in defaults
✖   subject may not be empty [subject-empty]
✖   type may not be empty [type-empty
```

**Comment by Yoda-BZH** [comment]
I was waiting for an approval before rebasing.

Thank you

**Review by noelmcloughlin — CHANGES_REQUESTED** [review]
This is approved and can be merged after commit message is updated.  

This formula has no code owner  so I'm just reviewing as passer by.  https://github.com/saltstack-formulas/node-formula/blob/c7e795783b5d158352857f276bdb86f8658617b5/CODEOWNERS#L9

**Comment by noelmcloughlin** [comment]
CI is passing so merging.  Thanks @Yoda-BZH

**Comment by myii** [comment]
@Yoda-BZH @noelmcloughlin Need a bit of help here because this PR has caused a regression in the CI for Debian-based platforms.  Unfortunately, the `packages-formula` CI also uses this formula for Debian-based platforms, so those are failing as well now.

* https://gitlab.com/myii/node-formula/-/pipelines/353776999
* https://gitlab.com/myii/packages-formula/-/pipelines/353777282

```sls
                 ID: node-package-repo-pkgrepo-managed
           Function: pkgrepo.managed
               Name: deb https://deb.nodesource.com/node_.x buster main
             Result: False
            Comment: Failed to configure repo 'deb https://deb.nodesource.com/node_.x buster main': E: The repository 'https://deb.nodesource.com/node_.x buster Release' does not have a Release file.
            Started: 09:57:49.961756
           Duration: 7236.536 ms
            Changes:   
       ----------
                 ID: node-package-install-pkg-installed
           Function: pkg.installed
               Name: nodejs
             Result: False
            Comment: An error was encountered while installing package(s): E: The repository 'https://deb.nodesource.com/node_.x buster Release' does not have a Release file.
            Started: 09:57:57.209571
           Duration: 2261.848 ms
            Changes:  
```

The issue here is that this breaks our "no pillar" approach, where formulas should work with sensible defaults without needing to even configure a pillar.  In this case, this forces a user to add a pillar entry for the version they need.

A better approach could be to construct the `repo.name` URL in the actual SLS file instead (putting all the pieces together).  Then a user could still provide the `pkg.version` and have it automatically update the URL as well.

Do you have any suggestions on how we should proceed, if not my proposal above?

**Comment by Yoda-BZH** [comment]
Hello @myii,

Sorry for breaking the pipelines.

Can you check https://github.com/saltstack-formulas/node-formula/pull/59 ? I'm not sure if this is the right way to implement what you're proposing.

Thank you

**Pull request merged**
